CONTRIBUTING.md

```diff
@@ -1,4 +1,27 @@
-# Contributing to plotly.js
+# Contributing to Plotly.js
+
+### How do changes get made to Plotly.js?
+
+The basic architecture of Plotly.js is to accept JSON representation of figures that adhere to the figure schema and draw graphical representations of these figures. Adding new attributes to the schema or new allowable values to existing attributes and adding the associated drawing code generally comprises a **new feature**. This project has a strong commitment to backwards-compatibility,  so changing the graphical output for existing schema attributes and values, or changing the default value of an attribute, is generally only done as a mostly-backwards-compatible **bug fix**, for cases when the current graphical output is incorrect, non-sensical or otherwise very problematic.  Non-schema-related features and bug fixes are possible as well, usually around performance, security, bundling, function signatures etc, and generally follow the same principles.
```

Comment:
Great!
Let's add a link to JSON file format e.g. here https://en.wikipedia.org/wiki/JSON.

Reply:
Might want to explicitly call out how new trace types still fit into this pattern, something like:

> A **new feature** is composed of additions to the schema - adding new attributes, adding entire new trace types, or just adding new values to existing attributes - along with the associated drawing code.

Also we talk about "adding things to the schema" but it's not clear to me that folks - even those who get this far - have a clear concept of what that really means. Perhaps we can link to https://plotly.com/javascript/reference/index/ - which itself links back to `plot-schema.json` and gives some more useful detail.

Reply:
done!